tests/app/main/views/test_admin_manager.py

```diff
@@ -204,3 +230,143 @@ def test_successful_post_flashes(self, data_api_client, send_user_account_email)
+
+    def test_should_correctly_load_and_prefill_edit_status_question(self, data_api_client):
+        self.user_role = "admin-manager"
+        data_api_client.get_user.return_value = self.admin_user_to_edit
+
+        response = self.client.get("/admin/admin-users/2345/edit")
+        assert response.status_code == 200
+
+        document = html.fromstring(response.get_data(as_text=True))
+
+        assert document.xpath('//span[@class="question-heading"]')[2].text.strip() == "Status"
+
+        assert document.cssselect('#input-edit_admin_status-1')[0].label.text.strip() == "Active"
+        assert document.cssselect('#input-edit_admin_status-1')[0].checked
+
+        assert document.cssselect('#input-edit_admin_status-2')[0].label.text.strip() == "Suspended"
+        assert not document.cssselect('#input-edit_admin_status-2')[0].checked
+
+    @pytest.mark.parametrize("role,expected_code", [
+        ("admin-manager", 200),
+        ("admin", 403),
+        ("admin-ccs-category", 403),
+        ("admin-ccs-sourcing", 403),
+    ])
+    def test_get_page_should_only_be_accessible_to_admin_manager(self, data_api_client, role, expected_code):
+        self.user_role = role
+        data_api_client.get_user.return_value = self.admin_user_to_edit
+
+        response = self.client.get("/admin/admin-users/2345/edit")
+        actual_code = response.status_code
+        assert actual_code == expected_code, "Unexpected response {} for role {}".format(response.status_code, role)
+
+    @pytest.mark.parametrize("role,expected_code", [
+        ("admin-manager", 302),
+        ("admin", 403),
+        ("admin-ccs-category", 403),
+        ("admin-ccs-sourcing", 403),
+    ])
+    def test_post_page_should_only_be_accessible_to_admin_manager(self, data_api_client, role, expected_code):
+        self.user_role = role
+        data_api_client.get_user.return_value = self.admin_user_to_edit
+
+        response = self.client.post('/admin/admin-users/2345/edit')
+        actual_code = response.status_code
+        assert actual_code == expected_code, "Unexpected response {} for role {}".format(response.status_code, role)
+
+    def test_admin_manager_can_edit_admin_user_details(self, data_api_client):
+        self.user_role = "admin-manager"
+        data_api_client.get_user.return_value = self.admin_user_to_edit
+        response1 = self.client.post(
+            "/admin/admin-users/2345/edit",
+            data={
+                "edit_admin_name": "Lady Myria Lejean",
+                "edit_admin_permissions": "admin",
+                "edit_admin_status": "False"
+            }
+        )
+        assert response1.status_code == 302
+        assert_flashes(self, "reality.auditor@digital.cabinet-office.gov.uk has been updated", "message")
+        assert data_api_client.update_user.call_args_list == [mock.call(
+            "2345", name="Lady Myria Lejean", role="admin", active=False
+        )]
+
+        response2 = self.client.get(response1.location)
```

Comment:
Can we assert the `location` we've redirected to?

Reply:
Good idea, I've done that now :)